contracts/SinglePlayerCommit.sol

```diff
@@ -51,213 +61,470 @@ contract SinglePlayerCommit is Ownable {
+            userId: _userId,
+            exists: true
         });
 
-        // ...and add it to storage
         commitments[msg.sender] = commitment;
 
-        emit NewCommitment(msg.sender, allowedActivities[_activityKey].name, _goalValue, _startTime, endTime, _stake);
+        emit NewCommitment(
+            msg.sender, 
+            activities[_activityKey].name, 
+            _goalValue, 
+            _startTime, 
+            endTime, 
+            _stake);
 
         return true;
     }
 
-    function withdraw(uint256 amount) public returns (bool) {
-        console.log("Received call for withdrawing amount %s from sender %s", amount, msg.sender);
-        uint256 available = balances[msg.sender].sub(commitments[msg.sender].stake);
-        require(amount >= available, "SPC::withdraw - not enough balance available");
-
-        // remove from committer's balance
-        _changeCommitterBalance(amount, false);
-
-        require(token.transfer(msg.sender, amount), "SPC::withdraw - token transfer failed");
-
-        emit Withdrawal(msg.sender, amount);
-
-        return true;
-    }
+    /// @notice Wrapper function to deposit <token> and create commitment in one call
+    /// @param _activityKey Keccak256 hashed, encoded name of activity
+    /// @param _goalValue Distance of activity as goal
+    /// @param _startTime Starttime of commitment, also used for endTime
+    /// @param _stake Amount of <token> to stake againt achieving goale
+    /// @param _depositAmount Size of deposit
+    /// @param _userId ???
+    /// @dev Call deposit and makeCommitment method
+    function depositAndCommit(
+        bytes32 _activityKey,
+        uint256 _goalValue,
+        uint256 _startTime,
+        uint256 _stake,
+        uint256 _depositAmount,
+        string memory _userId
+    ) 
+        public 
+        returns (bool success) 
+    {
+        require(deposit(_depositAmount), "SPC::depositAndCommit - deposit failed");
+        require(makeCommitment(
+                    _activityKey, 
+                    _goalValue, 
+                    _startTime, 
+                    _stake, 
+                    _userId
+                ), "SPC::depositAndCommit - commitment creation failed");
 
-    // TODO
-    function report() public returns (bool) {
-        // get activity data from oracle
-        // record activity data in commitments
         return true;
     }
 
+    //TODO DRY in procesCommitment methods
+    /// @notice Enables processing of open commitments after endDate that have not been processed by creator
+    /// @param committer address of the creator of the committer to process
+    /// @dev Process commitment by lookup based on address, checking metrics, state and updating balances
     function processCommitment(address committer) public {
         console.log("Processing commitment");
-        Commitment memory commitment = commitments[committer];
+        Commitment storage commitment = commitments[committer];
 
```

Comment:
First check on commitment.exists. Call from storage returns default object, so require() checks could lead to some null/0 comparisons

Reply:
done